flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -306,6 +319,40 @@ protected void resubmitJob(FlinkResourceContext<CR> ctx, boolean requireHaMetada
         restoreJob(ctx, specToRecover, ctx.getObserveConfig(), requireHaMetadata);
     }
 
+    private void redeployWithSavepoint(
+            FlinkResourceContext<CR> ctx,
+            Configuration deployConfig,
+            CR resource,
+            STATUS status,
+            SPEC currentDeploySpec,
+            JobState desiredJobState)
+            throws Exception {
+        LOG.info("Redeploying from savepoint");
+        cancelJob(ctx, UpgradeMode.STATELESS);
```

Comment:
So we ignore the upgrade policy here and discard the state? What if the new savepoint path is corrupted or doesn't exist?

Reply:
Users can always take a manual savepoint / set the state to suspended before performing a redeployment. But again redeploy is quite specific for cases when you actually have some unrecoverable error with your job during / after upgrade.

Reply:
I will add some extra docs